engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java

```diff
@@ -707,6 +715,34 @@ public void visit(@NotNull final Partition partition) {
                     groupByColumnNames));
         }
 
+        @Override
+        public void visit(@NotNull final Formula formula) {
+            final SelectColumn selectColumn =
+                    SelectColumn.of(Selectable.of(formula.column(), RawString.of(formula.formula())));
+
+            // Get or create a column definition map composed of vectors of the original column types.
+            if (vectorColumnNameMap == null) {
+                vectorColumnNameMap = new HashMap<>();
+                table.getColumnSourceMap().forEach((key, value) -> {
+                    final ColumnDefinition<?> columnDef = ColumnDefinition.fromGenericType(
+                            key, VectorFactory.forElementType(value.getType()).vectorType());
+                    vectorColumnNameMap.put(key, columnDef);
+                });
+            }
+
+            // Get the input column names from the formula and provide them to the groupBy operator
+            final String[] inputColumns =
+                    selectColumn.initDef(vectorColumnNameMap, compilationProcessor).toArray(String[]::new);
```

Comment:
why pay extra to turn this into a string array if you're about to stream it and turn _that_ into a MatchPair array? why not just do the map here directly?

Reply:
This is OBE (overcome by events) by additional changes.